pupil_src/shared_modules/video_capture/file_backend.py

```diff
@@ -220,6 +159,112 @@ def run_func(self, *args, **kwargs):
 
         return decorator
 
+    def _get_pattern_lst(self, source_path, timestamps=False):
+        '''
+        Get an order list follow the name pattern
+        '''
+        if timestamps:
+            pattern, ext = os.path.basename(
+                os.path.normpath(source_path)).split(".")
+            suffix = "_timestamps.npy"
+        else:
+            pattern, suffix = os.path.basename(
+                os.path.normpath(source_path)).split(".")
+            suffix = "." + suffix
+        path = []
+        for file in os.listdir(os.path.dirname(source_path)):
+            if fnmatch.fnmatch(file, pattern + "*" + suffix):
+                path.append(file)
+        return sorted(path)
+
+    def _get_conatiners_path(self, source_path):
+        return self._get_pattern_lst(source_path)
+
+    def _get_containers(self, index):
+        '''
+        Get container by index, find the next one if the prev one is broken.
+        '''
+        try:
+            container = av.open(self.containers_path[index])
+        except av.AVError:
+            logger.info("The vide is broken")
```

Comment:
```suggestion
            logger.info("The video is broken")
```
I think it would be nice if we could log that error once, including the file name, and then skip the broken files afterwards.

Reply:
Yes, I will update the log later.